StyleCop.Analyzers/StyleCop.Analyzers/MaintainabilityRules/SA1653UseTrailingCommasForMultiLineInitializers.cs

```diff
@@ -0,0 +1,77 @@
+    {
+        /// <summary>
+        /// The ID for diagnostics produced by the <see cref="SA1653UseTrailingCommasInMultiLineInitializers"/> analyzer.
+        /// </summary>
+        public const string DiagnosticId = "SA1653";
+        private static readonly LocalizableString Title = new LocalizableResourceString(nameof(MaintainabilityResources.SA1653Title), MaintainabilityResources.ResourceManager, typeof(MaintainabilityResources));
+        private static readonly LocalizableString MessageFormat = new LocalizableResourceString(nameof(MaintainabilityResources.SA1653MessageFormat), MaintainabilityResources.ResourceManager, typeof(MaintainabilityResources));
+        private static readonly LocalizableString Description = new LocalizableResourceString(nameof(MaintainabilityResources.SA1653Description), MaintainabilityResources.ResourceManager, typeof(MaintainabilityResources));
+        private static readonly string HelpLink = "https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1653.md";
+
+        private static readonly DiagnosticDescriptor Descriptor =
+            new DiagnosticDescriptor(DiagnosticId, Title, MessageFormat, AnalyzerCategory.ReadabilityRules, DiagnosticSeverity.Warning, AnalyzerConstants.EnabledByDefault, Description, HelpLink);
+
+        private static readonly Action<CompilationStartAnalysisContext> CompilationStartAction = HandleCompilationStart;
+        private static readonly Action<SyntaxNodeAnalysisContext> HandleObjectInitializerAction = HandleObjectInitializer;
+
+        /// <inheritdoc/>
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } =
+            ImmutableArray.Create(Descriptor);
+
+        /// <inheritdoc/>
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterCompilationStartAction(CompilationStartAction);
+        }
+
+        private static void HandleCompilationStart(CompilationStartAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleObjectInitializerAction, SyntaxKind.ObjectInitializerExpression);
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleObjectInitializerAction, SyntaxKind.AnonymousObjectCreationExpression);
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleObjectInitializerAction, SyntaxKind.ArrayInitializerExpression);
+            context.RegisterSyntaxNodeActionHonorExclusions(HandleObjectInitializerAction, SyntaxKind.CollectionInitializerExpression);
+        }
+
+        private static void HandleObjectInitializer(SyntaxNodeAnalysisContext context)
+        {
+            var initializer = (ExpressionSyntax)context.Node;
+            if (initializer == null
+                || !IsMultiline(initializer))
+            {
+                return;
+            }
+
+            var childNodesAndTokens = initializer.ChildNodesAndTokens().ToList();
```

Comment:
:thought_balloon: This can probably be handled more efficiently by registering two handlers and by using the appropriate property from `AnonymousObjectCreationExpressionSyntax` (for SyntaxKind.AnonymousObjectCreationExpression) and `InitializerExpressionSyntax` (for the others).